docs/designs/elasticdl_operator.md

```diff
@@ -127,73 +127,36 @@ spec:
   - "--num_minibatches_per_task 2"
   - "--evaluation_step 1000"
   Master:
-    template:
-      spec:
-        restartPolicy: Never
-        priorityClassName: high
-        containers:
-        - name: test-mnist-master
-          image: elasticdl-mnist
-          imagePullPolicy: Never
-          resources:
-            limits:
-              cpu: 1
-              memory: 2048Mi
-            requests:
-              cpu: 0.2
-              memory: 1024Mi
-          volumeMounts:
-          - mountPath: /data
-            name: test-volume
-        volumes:
-        - name: test-volume
-          hostPath:
-            path: /data
+    image: elasticdl-mnist
+    priority: high
+    resource_request:
+      cpu: 1
+      memory: 1024Mi
+    volume:
+      host_path=/host_data,mount_path=/data
   PS:
-    replicas: 2
-    template:
-      spec:
-        restartPolicy: Never
-        priorityClassName: high
-        containers:
-        - name: test-mnist-ps
-          image: elasticdl-ps
-          imagePullPolicy: Never
-          resources:
-            limits:
-              cpu: 1
-              memory: 2048Mi
-            requests:
-              cpu: 0.2
-              memory: 1024Mi
+    count: 2
+    priority: high
+    image: elasticdl-ps
+    resource_request:
+      cpu: 1
+      memory: 1024Mi
   Worker:
-    replicas: 10
-    template:
-      spec:
-        restartPolicy: Never
-        priorityClassName: low
-        containers:
-        - name: test-mnist-worker
-          image: elasticdl-mnist
-          imagePullPolicy: Never
-          resources:
-            limits:
-              cpu: 1
-              memory: 2048Mi
-            requests:
-              cpu: 0.2
-              memory: 1024Mi
-          volumeMounts:
-          - mountPath: /data
-            name: test-volume
-        volumes:
-        - name: test-volume
-          hostPath:
-            path: /data
+    count: 10
+    priority: high=0.5
+    image: elasticdl-worker
+    resource_request:
+      cpu: 4
+      gpu: 1
+      memory: 2048Mi
+    volume:
+      host_path=/host_data,mount_path=/data
```

Comment:
How about following the naming convention of Kubernetes pod spec.

```yaml
replicas: 10
priorityClassName: low
image: elasticdl-worker
resources:
  requests:
    cpu: 4
    gpu: 1
    memory: 2048Mi
volume: "host_path=/host_data,mount_path=/data"
```

Reply:
Controller will translate these values into ElasticDL arguments, such as:
`worker_pod_priority=low`, `ps_resource_request=cpu=1,memory=4096Mi`, and the master will use Kubernetes python API.
It is the Kubernetes cluster that determine the name of `priorityClassName`. And we want to support `high=0.5` format for workers. Also, ElasticDL does not care about resource limit `ps_resource_limit` as long as the requested resources are satisfied.

Thus, changed to:

```
  Worker:
    replicas: 10
    priority: high=0.5
    image: elasticdl-worker
    resource_request:
      cpu: 4
      gpu: 1
      memory: 2048Mi
    volume: "host_path=/host_data,mount_path=/data"
```